server/src/main/java/org/elasticsearch/reservedstate/ReservedClusterStateHandler.java

```diff
@@ -77,6 +77,20 @@ default Collection<String> dependencies() {
         return Collections.emptyList();
     }
 
+    /**
+     * List of optional dependent handler names for this handler.
+     *
+     * <p>
+     * These are dependent handlers which may or may not exist for this handler to be
+     * processed. If the optional dependency exists, then they are simply ordered to be
+     * merged into the cluster state before this handler.
+     *
+     * @return a collection of optional reserved state handler names
+     */
+    default Collection<String> optionalDependencies() {
```

Comment:
SLM and ILM might depend on this snapshot repo configuration, but not always. This extends the handler interface to allow optional dependencies, they are here simply for ordering purposes so we can create the repo before the SLM policy.